aii-core/src/main/perl/aii-web.cgi

```diff
@@ -79,9 +82,14 @@ sub Initialize {
   # find all profiles in directory
   my $d;
 
-
-  push @profiles,map { s/\.xml$//; $_.=".".$server_domain; } sort(grep(/\.xml$/, readdir(DIR)));    
-
+  # read GET and read profiles by extension (GET ?json)
+  if ($ENV{'REQUEST_METHOD'} eq "GET" and $ENV{'QUERY_STRING'}=='json') {
+    push @profiles,map { s/\.json$//; $_.=".".$server_domain; } sort(grep(/\.json$/, readdir(DIR)));
```

Comment:
factor out the common code (ie `json` or `xml`), and do the push only once. (this is too much copy/paste)